src/pages/countries.js

```diff
@@ -0,0 +1,82 @@
+import { useQuery, gql } from "@apollo/client";
+import { Link, useLocation, useHistory } from "react-router-dom";
+import CountriesList from "../components/CountriesList";
+import { useState, useEffect } from "react";
+
+export const COUNTRIES_QUERY = gql`
+  query allCountries($limit: Int, $skip: Int) {
+    countries(limit: $limit, skip: $skip) {
+      id
+      name
+      vatRate
+      languages {
+        id
+        name
+      }
+    }
+  }
+`;
+
+function useQueryUrl() {
+  return new URLSearchParams(useLocation().search);
+}
+
+const Countries = (props) => {
+  const query = useQueryUrl();
+  const history = useHistory();
+
+  const countriesPerPage = 5;
+  const [page, setPage] = useState(query.get("page"));
+
+  useEffect(() => {
+    setPage(query.get("page"));
+  }, [query.get("page")]);
+
+  const variables = {
+    variables: { limit: countriesPerPage, skip: page * countriesPerPage },
+  };
+
+  const { loading, error, data } = useQuery(COUNTRIES_QUERY, variables);
+
+  const getNextPageNumber = () => {
+    const currPage = parseInt(query.get("page")) || 0;
+    const nextPage = currPage + 1;
+    return nextPage;
+  };
+  const getPreviousPageNumber = () => {
+    const currPage = parseInt(query.get("page"));
+    if (currPage >= 1) {
+      const previousPage = currPage - 1;
+      return previousPage;
+    } else {
+      return 0;
+    }
+  };
+  const handleOnClickPrevious = () =>
+    history.push(`/countries?page=${getPreviousPageNumber()}`);
+  const handleOnClickNext = () =>
+    history.push(`/countries?page=${getNextPageNumber()}`);
+
+  if (loading) return <p>Loading...</p>;
+  if (error) return <p>Error :(</p>;
+    
+  return (
+    <div>
+      <p>Cinema page</p>
+      <div>
+        <button onClick={() => handleOnClickPrevious()} disabled={page <= 0}>
+          Wróć
+        </button>
+        <button onClick={() => handleOnClickNext()} disabled={page > 20}>
```

Comment:
I guess its only for demonstration purposes since we don't know how many pages it has in the `skip` scenario. I asked cause the logic of disabling next after reaching the last page number is a little bit weird, let's say that you have 20 pages, so in your logic it will disable it at 21 page which may not exist, but I guess it's the only demonstration disable so no worries 👍

Reply:
Same with handler as above